Source/WebKit/WebProcess/Extensions/API/Cocoa/WebExtensionAPIRuntimeCocoa.mm

```diff
@@ -201,35 +201,34 @@ - (instancetype)initWithAggregator:(WebKit::ReplyCallbackAggregator&)aggregator
     // Documentation: https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/runtime/getPlatformInfo
 
 #if PLATFORM(MAC)
-    static NSString * const osValue = @"mac";
+    static constexpr auto osValue = "mac"_s;
 #elif PLATFORM(IOS_FAMILY)
-    static NSString * const osValue = @"ios";
+    static constexpr auto osValue = "ios"_s;
 #else
-    static NSString * const osValue = @"unknown";
+    static constexpr auto osValue = "unknown"_s;
 #endif
 
 #if CPU(X86_64)
-    static NSString * const archValue = @"x86-64";
+    static constexpr auto archValue = "x86-64"_s;
 #elif CPU(ARM) || CPU(ARM64)
-    static NSString * const archValue = @"arm";
+    static constexpr auto archValue = "arm"_s;
 #else
-    static NSString * const archValue = @"unknown";
+    static constexpr auto archValue = "unknown"_s;
 #endif
 
-    static NSDictionary *platformInfo = @{
-        @"os": osValue,
-        @"arch": archValue
-    };
-
-    callback->call(platformInfo);
+    auto globalContext = callback->globalContext();
+    callback->call(fromObject(callback->globalContext(), {
+        { "os"_s, JSValueMakeString(globalContext, toJSString(osValue).get()) },
+        { "arch"_s, JSValueMakeString(globalContext, toJSString(archValue).get()) }
+    }));
 }
 
 void WebExtensionAPIRuntime::getBackgroundPage(Ref<WebExtensionCallbackHandler>&& callback)
 {
     // Documentation: https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/API/runtime/getBackgroundPage
 
     if (auto backgroundPage = protectedExtensionContext()->backgroundPage()) {
-        callback->call(toWindowObject(callback->globalContext(), *backgroundPage) ?: NSNull.null);
+        callback->call(toWindowObject(callback->globalContext(), *backgroundPage));
```

Comment:
`toWindowObject()` should return JS `null` instead of `undefined`, since that was happening here with `NSNull.null` before.